lib/mayaUsd/ufe/UsdRotatePivotTranslateUndoableCommand.cpp

```diff
@@ -25,8 +25,10 @@ UsdRotatePivotTranslateUndoableCommand::UsdRotatePivotTranslateUndoableCommand(c
 	, fPath(path)
 	, fNoPivotOp(false)
 {
-	conditionalCreateItem();
-
+	// create on first access
+	if(!fItem) {
+	    fItem = std::dynamic_pointer_cast<UsdSceneItem>(Ufe::Hierarchy::createItem(fPath));
+	}
```

Comment:
@ppt-adsk remove the duplicated conditionalCreateItem routine and calling it's logics directly in the constructor.

Reply:
Hmm, you sure about that?  The point during undo / redo of other commands, your fItem may go stale, so before you do your own undo or redo, you have to get a fresh one from the path.

I suggest renaming conditionalCreateItem() to just plain scene item, and it would look like an accessor:
UsdSceneItem sceneItem() const {
   if (!fItem) {
        fItem = std::dynamic_pointer_cast<UsdSceneItem>(Ufe::Hierarchy::createItem(fPath));
    }
    return fItem;
}
fItem would have to be made mutable.

Reply:
Good point, I totally missed that.... Still trying to wrap my head around the undo/redo command execution in various places:

addressed in : 7436e5a1c48a4270366ed100b311a57e86372112